### DISCUSSION TITLE
Supabase UUID Regex Pattern

### DISCUSSION THREAD

**Discussion by zfett** [supabase/supabase#2788] [Show and tell]
Hello all, I've made a regex pattern that can be used to validate Supabase UUIDs if needed. It even accounts for having multiple UUIDs on separate newlines! I've uploaded it over on RegExr, complete with tests! [Check it out here.](https://regexr.com/63eqf)

My regex is a bit rusty, so if you happen to notice that I forgot something or made a boo-boo, feel free to let me know and send any feedback my way!

---

Regex Pattern:
`/^([a-f]|[0-9]){8}-([a-f]|[0-9]){4}-([a-f]|[0-9]){4}-([a-f]|[0-9]){4}-([a-f]|[0-9]){12}$/gm`

**Comment by mozeryansky** [comment]
You don't need the group parenthesis and the character set syntax handles OR conditionals, so this could be:
`^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$`
Then you could use the digit character class:
`^[a-f\d]{8}-[a-f\d]{4}-[a-f\d]{4}-[a-f\d]{4}-[a-f\d]{12}$`
And also match the repeated pattern:
`^[a-f\d]{8}-([a-f\d]{4}-){3}[a-f\d]{12}$`

Also, the UUID standard does not specify the casing, so really it's valid uppercase or lowercase, which can be handled with `/i` to ignore the case instead of specifying the capital letters.
